Updated user sign-in/login and creation of user-session #17

gregtuc opened the pull request:
Several updates. Manual POST method in clients login.js was replaced with methods from Axios, allowing for an easier process. After signing in, session is now created with express-session - this will allow the user to remain signed in (see path /login in server index.js) which should enhance #17 , though additional features are needed. Basic sign-in page was also complete which closes #12 .
Very basic layout for a home page was also created, mostly for the purposes of future testing of redirection after sign-in.

gregtuc committed:
Making changes to express paths

Changing express paths to allow for basic user login

gregtuc committed:
Updated login

creates session and using axios to send post requests instead

gregtuc committed:
adding package-lock to gitignore

gregtuc committed:
hiding package-lock.json in gitignore

gregtuc committed:
Update .gitignore

gregtuc committed:
Update .gitignore

gregtuc committed:
Removing as not necessary

gregtuc committed:
removing unnecessary packagelockjson

gregtuc committed:
Updating README.md with frameworks/languages

ginywiny commented on server/index.js:
```diff
@@ -1,57 +1,80 @@
+const mongoclient = require("mongodb").MongoClient;
+const objectid = require("mongodb").ObjectID;
+const uri =
+  "mongodb+srv://greg34910:cookie123@cluster0-pf0b2.mongodb.net/test?retryWrites=true&w=majority";
```
You are adding your personal mongoDB account info. Can you make this generic?

gregtuc replied:
Oops! Nice catch, will change that.

ginywiny commented:
Don't forget to reference commits with issue numbers, so we can reference the issue if ever there are multiple issues worked on for one pull request

gregtuc committed:
removing personal mongodb info

gregtuc commented:
> Don't forget to reference commits with issue numbers, so we can reference the issue if ever there are multiple issues worked on for one pull request

Two separate issues should already be referenced in the description

gregtuc committed:
Creation of a home page #34

A home page that will be used for testing purposes - code can be re-used/adapted for creation of a news feed moving forward.

gregtuc committed:
Merge branch 'master' of https://github.com/gregtuc/Soen-341
